scripts/dependency_graph.py

```
@@ -0,0 +1,109 @@
+    def __init__(self, errors: list[tuple[str, str]]):
+        self.errors =errors
+    def __repr__(self):
+        msg = [str(self.__class__)]
+        for app_name, dep_name in self.errors:
+            msg.append(f"Dependency({dep_name}) in App({app_name}) is invalid")
+        return os.linesep.join(msg)
+
+def load_dependencies() -> dict[str, DependencyFile]:
+    """Load the JSON file containing application dependencies."""
+    results = {}
+    for dirpath, dirnames, filenames in pathlib.Path("kubernetes/main/apps").resolve().walk():
```

Ruff (0.14.4)

Loop control variable `dirnames` not used within loop body

Rename unused `dirnames` to `_dirnames`

(B007)